src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,398 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    WeilerAtherton()
```

Comment:
```suggestion
    WeilerAthertonClipping()
```

Reply:
All done, rebased and fixed the method naming.

Reply:
Thank you @jnemanja ❤️ will reserve some time to review the PR carefully now that it is cleaned and rebased. We will need some additional iterations, but the code is getting much easier to digest now.